Lib/importlib/metadata.py

```diff
@@ -4,20 +4,24 @@
 import csv
 import sys
 import email
+import inspect
 import pathlib
 import zipfile
 import operator
+import warnings
 import functools
 import itertools
 import posixpath
-import collections
+import collections.abc
+
+from ._itertools import unique_everseen
 
 from configparser import ConfigParser
 from contextlib import suppress
 from importlib import import_module
 from importlib.abc import MetaPathFinder
 from itertools import starmap
-from typing import Any, List, Optional, Protocol, TypeVar, Union
+from typing import Any, List, Mapping, Optional, Protocol, TypeVar, Union
 
 
 __all__ = [
```

Comment:
This can be updated with `packages_distributions` as a public function.

Reply:
Added to https://github.com/python/importlib_metadata/commit/7bdeaa45e3710e33735a9632c65ab7916c06c410, to be included in a future sync.